src/app/proptypes.js

```diff
@@ -0,0 +1,41 @@
+import propTypes from 'prop-types';
+
+export const imagePropTypes = {
+  model: propTypes.shape({
+    blocks: propTypes.arrayOf(
+      propTypes.shape({
+        locator: propTypes.string,
+      }),
+    ),
+  }),
+};
+
+export const textPropTypes = {
+  blocks: propTypes.arrayOf(
+    propTypes.shape({
+      model: propTypes.shape({
+        blocks: propTypes.arrayOf(
+          propTypes.shape({
+            text: propTypes.string,
+          }),
+        ),
+      }),
+    }),
+  ),
+};
+
+const baseDefaultPropTypes = {
+  model: {
+    blocks: [
+      {
+        model: {},
+      },
+    ],
+  },
+};
+
+export const imageDefaultPropTypes = baseDefaultPropTypes;
+
+export const textDefaultPropTypes = {
+  blocks: [baseDefaultPropTypes],
+};
```

Comment:
Is there a way to generalize these even more? By that I mean splitting them into small containarised blocks that you could stich together to your liking in the actual test? Sort of like a generic way of buyilding the props for all tests?

Reply:
@DemonusPC  Yes, that is possible - however for all the existing components, these are sufficient. We can abstract out the block creation if there is more variation in future components.

Reply:
Alright, all of this is great. The modularity of this as it grows could be its own seperate issue.